src/rgw/rgw_compression.cc

```diff
@@ -0,0 +1,164 @@
+
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "rgw_compression.h"
+
+#define dout_subsys ceph_subsys_rgw
+
+//------------RGWPutObj_Compress---------------
+
+int RGWPutObj_Compress::handle_data(bufferlist& bl, off_t ofs, void **phandle, rgw_obj *pobj, bool *again)
+{
+  bufferlist in_bl;
+
+  // compression stuff
+  if ((ofs > 0 && compressed) ||                                // if previous part was compressed
+      (ofs == 0)) {                                             // or it's the first part
+    ldout(cct, 10) << "Compression for rgw is enabled, compress part " << bl.length() << dendl;
```

Comment:
in testing, i'm always seeing `Compression for rgw is enabled, compress part 0` at the end of an upload. can you add a check for len == 0 somewhere, to avoid storing compression_blocks with len=0?

i'm not sure whether that's best done here in `RGWPutObj_Compress`, or in the do-while loop in `RGWPutObj::execute()`